website_supplier_list/controllers/main.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# © 2016 Michael Viriyananda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import SUPERUSER_ID
+from openerp.addons.web import http
+from openerp.addons.website.models.website import unslug
+from openerp.tools.translate import _
+from openerp.addons.web.http import request
+import werkzeug.urls
+
+
+class WebsiteSupplier(http.Controller):
+    _references_per_page = 20
+
+    @http.route([
+        '/suppliers',
+        '/suppliers/page/<int:page>',
+        '/suppliers/country/<int:country_id>',
+        '/suppliers/country/<country_name>-<int:country_id>',
+        '/suppliers/country/<int:country_id>/page/<int:page>',
+        '/suppliers/country/<country_name>-<int:country_id>/page/<int:page>',
+    ], type='http', auth="public", website=True)
+    def suppliers(self, country_id=0, page=0, country_name='', **post):
+        cr, uid, context = request.cr, request.uid, request.context
+        country_obj = request.registry['res.country']
+        partner_obj = request.registry['res.partner']
+        partner_name = post.get('search', '')
+
+        domain = [
+            ('website_supplier_published', '=', True),
+            ('supplier', '=', True)
+        ]
+        if partner_name:
+            domain += [
+                '|',
+                ('name', 'ilike', post.get("search")),
+                ('website_description', 'ilike', post.get("search"))
+            ]
+
+        # group by country, based on suppliers found with the search(domain)
+        countries = partner_obj.read_group(
+            cr, SUPERUSER_ID, domain, ["id", "country_id"],
```

Comment:
I prefer an ACL for the public user that allows him to read partners that are suppliers and are published, so you don't have to use `SUPERUSER_ID`.

BTW, use new api please.